WindowBoard.cpp

```diff
@@ -143,6 +152,24 @@ GameWindow::showBoard()
 	fScore->SetText(score.String());
 }
 
+void
+GameWindow::FrameResized(float width, 
+			 float height)
+{
+	float ratio = width/prevWidth;
+	SetScale(ratio);
+	prevWidth = width;
+	prevHeight = height;
+	// Keep everything in scale and in the right width:height ratio
+	ResizeTo(width, height*ratio);
+	// We don't want the user to scale the window so small that
+	// there's no space for the buttons.
+	if (width < defaultWidth) {
+		ResizeTo(defaultWidth, defaultHeight);
```

Comment:
When this happens, you will be calling ResizeTo twice, before and inside the if. Can you rearrange the code so that this does not happen? We want to resize only once when we are sure we got the final size.

Reply:
Oops. You're right. I'll fix this right away :)